### PR TITLE
Reworks the jukebox to use the CDN (if available). Please ensure all your jukebox music is .mp3, and that you use a CDN. Looking at you, Skyrat.

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request

Reworks the jukebox to use the CDN (if available). Please ensure all your jukebox music is .mp3.

If you don't have the CDN enabled you're not going to see the lag fix from this. Looking at you, Skyrat. Please setup a CDN already and use the webroot asset transport system.

## Why It's Good For The Game

This fixes a metric fuckton of the lag with the Jukebox, aka the disco ball on Uncle Pete's Rollerdome. It should no longer murder people when it plays Caramelldansen, and now it'll be synced for everyone on the server, even if you walk out of the range of the disco ball and walk back into it.

## Changelog

:cl:
refactor: Uncle Pete's Rollerdome has upgraded it's jukebox software, and now it's more consistent and way, way faster for people with slower computers or internet connections!
/:cl:

### CODE REVIEW

File: code/game/machinery/dance_machine.dm [hunk 1]
```diff
@@ -12,12 +12,14 @@
...
-	var/volume = 50
+	var/volume = 100
+	var/current_song_url
+	var/asset_md5_cache
 	COOLDOWN_DECLARE(jukebox_error_cd)
 
 /obj/machinery/jukebox/disco
 	name = "radiant dance machine mark IV"
-	desc = "The first three prototypes were discontinued after mass casualty incidents."
+	desc = "The first three prototypes were discontinued after mass casual 	ty incidents."
```
Comment (SuperNovaa41) [inline code/game/machinery/dance_machine.dm:22]: ?

File: code/game/machinery/dance_machine.dm [hunk 2]
```diff
@@ -174,6 +178,46 @@
...
+	var/datum/asset_cache_item/ACI = new(asset_md5, selection.song_path)
+	ACI.namespace = "jukebox_muzak"
+	if(istype(AS, /datum/asset_transport/webroot))
+		var/datum/asset_transport/webroot/WR = AS
+		WR.save_asset_to_webroot(ACI)
+	else
+		AS.register_asset(asset_md5, ACI)
+		asset_md5_cache = asset_md5
+	current_song_url = AS.get_asset_url(null, ACI)
+	for(var/m in GLOB.player_list)
```
Comment (SuperNovaa41) [inline code/game/machinery/dance_machine.dm:192]: use better var names

File: code/game/machinery/dance_machine.dm (hunk 1 above)
Comment (Iamgoofball) [reply code/game/machinery/dance_machine.dm:22]: whoops

File: code/game/machinery/dance_machine.dm (hunk 2 above)
Comment (Iamgoofball) [reply code/game/machinery/dance_machine.dm:192]: ok

File: code/game/machinery/dance_machine.dm [hunk 3]
```diff
@@ -119,7 +123,7 @@
 		data["track_selected"] = selection.song_name
 		data["track_length"] = DisplayTimeText(selection.song_length)
 		data["track_beat"] = selection.song_beat
-	data["volume"] = volume
+	data["volume"] = min(volume, 100) // funny hrefsploit
```
Comment (Fikou) [inline code/game/machinery/dance_machine.dm:126]: this is ui data what exploit are you gonna get here

Comment (Iamgoofball) [reply code/game/machinery/dance_machine.dm:126]: lmao sorry wrong spot

**Review by MrsTonedOne — CHANGES_REQUESTED** [review]
As a note, our cdn has been disabled for a spell while I work out some issues.

Please make sure I have taken steps to fix this before merging.

Thanks.

**Comment by Iamgoofball** [comment]
@MrStonedOne can you slap a DNM and a RED LABEL on this one then?

**Comment by LemonInTheDark** [comment]
He can't ~~the weak should fear the strong~~

File: code/game/machinery/dance_machine.dm (hunk 3 above)
Comment (LemonInTheDark) [reply code/game/machinery/dance_machine.dm:126]: lol

**Comment by MrStonedOne** [comment]
> @MrStonedOne can you slap a DNM and a RED LABEL on this one then?

I, like you, lost my issue jannie status because i fucked with prs.

**Comment by stylemistake** [comment]
@Iamgoofball for you: #65943

**Comment by tralezab** [comment]
conflict

**Pull request closed without merging**